go/exorelay/exorelay_test.go

```diff
@@ -27,16 +29,51 @@ func newExocom() *exocomMock.ExoComMock {
 	return exocom
 }
 
+func WaitFor(condition func() bool, errorMessage string) error {
+	success := make(chan bool, 1)
+
+	go func() {
+		for {
+			time.Sleep(time.Millisecond)
+			if condition() {
+				success <- true
+				break
+			}
+		}
+	}()
+
+	select {
+	case <-success:
+		return nil
+	case <-time.After(time.Second * 10):
+		return errors.New(errorMessage)
+	}
+}
+
+func WaitForReceivedMessagesCount(exocom *exocomMock.ExoComMock, count int) error {
```

Comment:
We could use https://github.com/matryer/try instead of rolling our own WaitFor helper.